logicalplan/plan.go

```diff
@@ -131,6 +133,112 @@ func traverseBottomUp(parent *parser.Expr, current *parser.Expr, transform func(
 	return true
 }
 
+// preprocessExpr wraps all possible step invariant parts of the given expression with
+// StepInvariantExpr. It also resolves the preprocessors.
+func preprocessExpr(expr parser.Expr, start, end time.Time) parser.Expr {
```

Comment:
I'm guessing this had to be copied over, to work with `internal/prometheus/parser`? Might be worth adding a comment for that! 🙂

Reply:
Added